bundles/com.zeligsoft.domain.dds4ccm/src/com/zeligsoft/domain/dds4ccm/utils/DDS4CCMMigrationModelTypeUtil.java

```diff
@@ -69,35 +92,268 @@ private static boolean migrateAll(Model model, boolean dryrun) {
 
-			if(ZDLUtil.isZDLConcept(e, CCMNames.RESOURCE)){
-				Resource resource = ZDLFactoryRegistry.INSTANCE.create(e, Resource.class);
-				List<String> oldStrings = new ArrayList<String>(resource.getResourceType());
-				// ZDLUtil is used for setting value due to the limitation of the api code
-				ZDLUtil.setValue(e, CCMNames.REQUIREMENT_SATISFIER, CCMNames.REQUIREMENT_SATISFIER__RESOURCE_TYPE, null);				
+			migrateToAxciomaProperty(e);
+			
+			// load default port connectorTypes
+			if(ZDLUtil.isZDLConcept(e, CCMNames.INTERFACE_PORT)){
+				InterfacePort ip = ZDLFactoryRegistry.INSTANCE.create(e, InterfacePort.class);
+				
+				if(!ZDLUtil.isZDLConcept(ip.getPorttype().eObject(), CORBADomainNames.CORBAINTERFACE)){
+					continue;
+				}
+				
+				CORBAInterface portType = ZDLFactoryRegistry.INSTANCE.create(ip.getPorttype().eObject(), CORBAInterface.class);
+				
+				boolean isPortTypeLocal = portType.getIsLocal();
 				
-				for(String s : oldStrings) {
-					resource.addResourceType(getAxciomaStr(s));
+				boolean isNonLocalAsyncUsesPort = (!isPortTypeLocal && ip.getIsAsynchronous()) && ip.getIsConjugated();
+				boolean isNonLocalAsyncProvidesPort = (!isPortTypeLocal && ip.getIsAsynchronous()) && !ip.getIsConjugated();
+				
+				boolean isNonLocalSyncUsesPort = (!isPortTypeLocal && !ip.getIsAsynchronous()) && ip.getIsConjugated();
+				boolean isNonLocalSyncProvidesPort = (!isPortTypeLocal && !ip.getIsAsynchronous()) && !ip.getIsConjugated();
+				
+				if(isNonLocalAsyncUsesPort){
+					ip.setConnectorType(connectorDefAsyncUses);
+				}else if(isNonLocalAsyncProvidesPort){
+					ip.setConnectorType(connectorDefAsyncProvides);
+				}else if(isNonLocalSyncUsesPort){
+					ip.setConnectorType(connectorDefSyncUses);
+				}else if(isNonLocalSyncProvidesPort){
+					ip.setConnectorType(connectorDefSyncProvides);
 				}
-			}else if(ZDLUtil.isZDLConcept(e, CCMNames.PROPERTY)){
-				com.zeligsoft.domain.omg.ccm.api.CCM_Target.Property oldProperty =  ZDLFactoryRegistry.INSTANCE.create(e, com.zeligsoft.domain.omg.ccm.api.CCM_Target.Property.class);
-				oldProperty.setName(getAxciomaStr(oldProperty.getName()));
-			}else if(ZDLUtil.isZDLConcept(e, CCMNames.SATISFIER_PROPERTY)){
-				SatisfierProperty oldSatisfierProperty =  ZDLFactoryRegistry.INSTANCE.create(e, SatisfierProperty.class);
-				ZDLUtil.setValue(e, CCMNames.SATISFIER_PROPERTY, CCMNames.SATISFIER_PROPERTY__VALUE, getAxciomaStr((String)ZDLUtil.getValue(e, CCMNames.SATISFIER_PROPERTY, CCMNames.SATISFIER_PROPERTY__VALUE)));
-				oldSatisfierProperty.setName(getAxciomaStr(oldSatisfierProperty.getName()));
 			}
 		}
 	}
 	
+	/**
+	 * Given a connectorType for 'uses' port, returns corresponding connectorType for 'provides' port    
+	 * 
+	 * @param defaultConnectorTypeUses
+	 *         	default connectorType set in preferences. To be used in the 'uses' port during migration  
+	 *          
+	 */
+	private static String getDefaultConnectorTypeProvides(String defaultConnectorTypeUses){
```

Comment:
Another case where the knowledge of the number of Axcioma connector types is diffused throughout the code. See my comment below.